### PR TITLE
Add Vault kv scoped storage logic

### PR DESCRIPTION
**Pull request by vshatravenko**

### CODE REVIEW

File: pkg/vault/vault.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	}
+
+	return nil
+}
+
+// SetSecret stores all secrets into the memory
+func (vs *Service) SetSecret(name string, value interface{}, scope string) error {
+	// Secret scope only supports strings
+	if scope == "secret" {
+		encrypted, err := vs.Encrypt(fmt.Sprintf("%v", value))
```
Comment (calj) [inline pkg/vault/vault.go:154]: better to check that this is actually a string:
`
str, ok := values.(string)
if !ok {
return fmt.Errors("")
}
`

File: pkg/vault/vault.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	if vs.data == nil {
+		vs.data = make(map[string]interface{})
+	}
+
+	if secret == nil || secret.Data == nil {
+		// fmt.Println("Secret or data is nil, overwriting:", vs.data[scope])
+		vs.data[scope] = make(map[string]interface{})
+	} else {
+		// fmt.Println("Secret is not nil, writing", secret.Data["data"])
+		vs.data[scope] = secret.Data["data"].(map[string]interface{})
```
Comment (jengjeng) [inline pkg/vault/vault.go:144]: seems `secret.Data["data"]` will be a string and can be json ? should parse it into interface.

Comment (vshatravenko) [reply pkg/vault/vault.go:144]: It's a map

Comment (vshatravenko) [reply pkg/vault/vault.go:144]: map[string]interface{}

File: pkg/config/config.go [hunk 3]
```diff
@@ -1,29 +1,45 @@
...
+	"github.com/openware/kaigara/types"
 )
 
+// KaigaraConfig contains cli options
+type KaigaraConfig struct {
+	SecretStore  string `yaml:"secret-store" env:"KAIGARA_SECRET_STORE" env-default:"vault"`
+	VaultToken   string `yaml:"vault-token" env:"KAIGARA_VAULT_TOKEN"`
+	VaultAddr    string `yaml:"vault-addr" env:"KAIGARA_VAULT_ADDR" env-default:"http://127.0.0.1:8200"`
+	AppName      string `yaml:"vault-app-name" env:"KAIGARA_APP_NAME"`
+	DeploymentID string `yaml:"deployment-id" env:"KAIGARA_DEPLOYMENT_ID"`
```
Comment (dnfd) [inline pkg/config/config.go:18]: +1 vote against dashes

File: Makefile [hunk 4]
```diff
@@ -1,6 +1,7 @@
 build:
 	go build -o bin/kaigara cmd/kaigara/kaigara.go
 	go build -o bin/kaitail cmd/kaitail/kaitail.go
+	go build -o bin/kaisave cmd/kaisave/kaisave.go
```
Comment (calj) [inline Makefile:4]: better to use the syntax
```
	go build -o bin/kaisave ./cmd/kaisave/
```
so we can potentially put more go files in the directory

File: cmd/kaigara/kaigara.go [hunk 5]
```diff
@@ -72,41 +98,47 @@ func kaigaraRun(ls logstream.LogStream, cnf config.Config, channelName, cmd stri
...
-	return config.NewVaultConfig(addr, token, path)
-}
-
 func initLogStream() logstream.LogStream {
 	url := os.Getenv("KAIGARA_REDIS_URL")
 	return logstream.NewRedisClient(url)
 }
 
+func exitWhenSecretsOutdated(c *exec.Cmd, secretStores []types.SecretStore, scopes []string) {
+	for range time.Tick(time.Minute * 1) {
```
Comment (calj) [inline cmd/kaigara/kaigara.go:107]: maybe we put 20 seconds

**Pull request merged**